tests/test_sockets.py

```diff
@@ -1838,3 +1838,14 @@ async def test_getnameinfo() -> None:
     expected_result = socket.getnameinfo(("127.0.0.1", 6666), 0)
     result = await getnameinfo(("127.0.0.1", 6666))
     assert result == expected_result
+
+
+async def test_connect_tcp_getaddrinfo_context() -> None:
+    """
+    See https://github.com/agronholm/anyio/discussions/580
```

Comment:
Pending the discussion->issue conversion as mentioned above.

Reply:
```suggestion
    See https://github.com/agronholm/anyio/issues/815
```